src/hooks/useQuery.ts

```diff
@@ -33,19 +33,34 @@ export type UseQueryResponse<T> = [
 export const useQuery = <T = any, V = object>(
   args: UseQueryArgs<V>
 ): UseQueryResponse<T> => {
+  const isMounted = useRef(false);
   const unsubscribe = useRef(noop);
+  const initialState = useRef<UseQueryState<T>>({
+    data: undefined,
+    error: undefined,
+    fetching: true,
+  });
 
   const client = useContext(Context);
+
   const request = useMemo(
     () => createRequest(args.query, args.variables as any),
     [args.query, args.variables]
   );
 
-  const [state, setState] = useState<UseQueryState<T>>({
-    fetching: false,
```

Comment:
Can we simplify this PR to only change the initial state?

This line being changed from `fetching: false` to `fetching: args.pause !== true` would set the initial state as fetching by default if pause is not specified.

Reply:
`args.pause !== true` is in theory not necessary anymore if we handle this correctly via `executeQuery` being either active or not active after it's run.

I suppose this PR is still in a more "experimental" stage so maybe we can merge yours for now and if that leads to conflicts I can later resolve them 🤔

Reply:
> args.pause !== true is in theory not necessary anymore if we handle this correctly via executeQuery being either active or not active after it's run.

Adding this logic to executeQuery would not solve the problem of the initial rendering state though?

Reply:
Talking about pause this should probably also be accounted in first render trying to get from cache else we’d make that option obsolete for first call.